client/profile-wizard/steps/benefits/test/index.js

```diff
@@ -0,0 +1,205 @@
+		const { container, queryByText } = render( <BenefitsLayout /> );
+
+		expect(
+			container.querySelectorAll(
+				'.woocommerce-profile-wizard__benefit-card'
+			).length
+		).toBe( 3 );
+		expect( queryByText( 'Store management on the go' ) ).not.toBeNull();
+		expect( queryByText( 'Automated sales taxes' ) ).not.toBeNull();
+		expect( queryByText( 'Improved speed & security' ) ).not.toBeNull();
+
+		useSelect.mockImplementation( () => ( {
+			activePlugins: [ 'jetpack', 'woocommerce-services' ],
+			isJetpackConnected: false,
+		} ) );
+
+		expect(
+			container.querySelectorAll(
+				'.woocommerce-profile-wizard__benefit-card'
+			).length
+		).toBe( 3 );
+		expect( queryByText( 'Store management on the go' ) ).not.toBeNull();
+		expect( queryByText( 'Automated sales taxes' ) ).not.toBeNull();
+		expect( queryByText( 'Improved speed & security' ) ).not.toBeNull();
+	} );
+
+	it( 'should show Jetpack benefits when Jetpack is not connected', () => {
+		useSelect.mockImplementation( () => ( {
+			activePlugins: [ 'jetpack', 'woocommerce-services' ],
+			isJetpackConnected: false,
+		} ) );
+
+		const { container, queryByText } = render( <BenefitsLayout /> );
+
+		expect(
+			container.querySelectorAll(
+				'.woocommerce-profile-wizard__benefit-card'
+			).length
+		).toBe( 3 );
+		expect( queryByText( 'Store management on the go' ) ).not.toBeNull();
+		expect( queryByText( 'Automated sales taxes' ) ).not.toBeNull();
+		expect( queryByText( 'Improved speed & security' ) ).not.toBeNull();
+	} );
+
+	it( 'should skip the benefits step when setup is already complete', () => {
+		useSelect.mockImplementation( () => ( {
+			activePlugins: [ 'jetpack', 'woocommerce-services' ],
+			isJetpackConnected: true,
+		} ) );
+		const goToNextStep = jest.fn();
+
+		const { container } = render(
+			<BenefitsLayout goToNextStep={ goToNextStep } />
+		);
+
+		expect( container ).toBeEmptyDOMElement();
+		expect( goToNextStep ).toHaveBeenCalled();
+	} );
+
+	it( 'should skip the plugin installation when clicking skip', () => {
+		const goToNextStep = jest.fn();
+		const updateProfileItems = jest.fn();
+
+		useSelect.mockImplementation( () => ( {
+			activePlugins: [],
+			isJetpackConnected: false,
+		} ) );
+		useDispatch.mockReturnValue( {
+			updateProfileItems,
+		} );
+
+		const { queryByText } = render(
+			<BenefitsLayout goToNextStep={ goToNextStep } />
+		);
+
+		fireEvent.click( queryByText( 'No thanks' ) );
+		expect( updateProfileItems ).toHaveBeenCalledWith( {
+			plugins: 'skipped',
+		} );
+	} );
+
+	it( 'should install the plugins when opting in', async () => {
+		const goToNextStep = jest.fn();
+		const updateProfileItems = jest.fn();
+		const installAndActivatePlugins = jest.fn();
+
+		useSelect.mockImplementation( () => ( {
+			activePlugins: [],
+			isJetpackConnected: false,
+		} ) );
+		useDispatch.mockReturnValue( {
+			installAndActivatePlugins,
+			updateOptions: jest.fn(),
+			updateProfileItems,
+		} );
+
+		await act( async () => {
```

Comment:
Not a big deal but are you sure you need `act` here? you've mocked everything so its gonna be completely synchronous, in theory this shouldn't be required?

Unless this is related to how wordpress/data handles dispatch?

Reply:
I was adding `act` due to React warnings.  Are you saying those can be ignored?

Reply:
@joshuatf nah all good. I know some people just add act around things, so I wanted to check. This is :+1:

Reply:
Gotcha.  Good looking out.

`act( () => merge() );`